rfcs/text/semantic-release.md

```diff
@@ -0,0 +1,257 @@
+> [[Source]](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit)
+
+We will use [`commitlint`](https://github.com/conventional-changelog/commitlint) and [`husky`](https://github.com/typicode/husky) to check our commits against the `conventionalcommits` guidelines.
+- [`husky`](https://github.com/typicode/husky) will manage the git commit hooks defined in `package.json`
+- we will write a hook that runs the commit message via [`commitlint`](https://github.com/conventional-changelog/commitlint)
+- we will use the [`@commitlint/config-conventional`](https://github.com/conventional-changelog/commitlint/tree/master/%40commitlint/config-conventional) config
+- [`husky`](https://github.com/typicode/husky) will abort the commit if the commit message is in the wrong format
+
+We will use [`commitizen`](https://github.com/commitizen/cz-cli) to help write commit messages
+- we will use the [`cz-conventional-changelog`](https://github.com/commitizen/cz-conventional-changelog) config
+- users can choose to write commit messages with the regular `git` CLI or in their IDE still
+- users can launch [`commitizen`](https://github.com/commitizen/cz-cli) with `npm run commit`
+
+We will add [`probot/semantic-pull-requests`](https://probot.github.io/apps/semantic-pull-requests/) as a GitHub check.
+This will prevent PR's being merged that do not follow the correct syntax.
+
+## Reverting a Release
+> If you have introduced a breaking bug in a release you have 2 options:
+> * If you have a fix immediately ready, commit and push it (or merge it via a pull request) to the release branch
+> * Otherwise, [revert the commit](https://git-scm.com/docs/git-revert) that introduced the bug and push the revert commit (or merge it via a pull request) to the release branch  
+> 
+> In both cases **semantic-release** will publish a new release, so your package users will get the fixed/reverted version.
+>
+> Depending on the package manager you are using, you might be able to un-publish or deprecate a release, in order to prevent users from downloading it by accident. For example, [NPM](https://www.npmjs.com/) allows you to [un-publish](https://docs.npmjs.com/cli/unpublish) [within 72 hours](https://www.npmjs.com/policies/unpublish) after release. You may also [deprecate](https://docs.npmjs.com/cli/deprecate) a release if you would rather avoid un-publishing.
+>
+> In any case **do not remove the Git tag associated with the buggy version**, otherwise **semantic-release** will later try to republish that version. Publishing a version after un-publishing is not supported by most package managers.  
+> [[Source]](https://semantic-release.gitbook.io/semantic-release/support/faq#how-can-i-revert-a-release)
+
+## Release Notes
+
+In version 4.3.0 we introduced [`renogen`](https://github.com/DDAZZA/renogen) a ruby gem to allow us to write release notes as a collection of changes. Prior
+to this I believe they were written and updated by each collaborator.
```

Comment:
I think this is how it used to be done:
```suggestion
to this, the release notes were written and updated by each collaborator manually in the old `CHANGELOG.md` file.
```
For reference, see the blame of the old `CHANGELOG.md` file:

* https://github.com/Sage/carbon/blame/v4.2.0/CHANGELOG.md